js/battle.js

```diff
@@ -524,18 +524,21 @@ var Pokemon = (function () {
 			this.baseAbility = ability;
 		}
 	};
+	Pokemon.prototype.prepName = function (name) {
```

Comment:
Mainly because neither `this.name` or `this.species` would be containing the entire nickname with parenthesis or `<small>` tags (`Nickname <small>(species)</small>`). Passing in the already constructed name seemed better than reconstructing the nickname again.